compiler/rustc_hir_typeck/src/cast.rs

```diff
@@ -865,7 +892,15 @@ impl<'a, 'tcx> CastCheck<'tcx> {
                         ));
 
                         // `dyn Src = dyn Dst`, this checks for matching traits/generics
-                        fcx.demand_eqtype(self.span, src_obj, dst_obj);
+                        let cause = fcx.misc(self.span);
```

Comment:
Leave a comment saying that this is just `demand_eqtype` but inlined so it has a better error